code/datums/armor.dm

```diff
@@ -0,0 +1,66 @@
+#define ARMORID "armor-[melee]-[bullet]-[laser]-[energy]-[bomb]-[bio]-[rad]-[fire]-[acid]"
+
+/proc/getArmor(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+  . = locate(ARMORID)
+  if (!.)
+    . = new /datum/armor(melee, bullet, laser, energy, bomb, bio, rad, fire, acid)
+
+/datum/armor
+	var/melee
+	var/bullet
+	var/laser
+	var/energy
+	var/bomb
+	var/bio
+	var/rad
+	var/fire
+	var/acid
+
+/datum/armor/New(melee_value = 0, bullet_value = 0, laser_value = 0, energy_value = 0, bomb_value = 0, bio_value = 0, rad_value = 0, fire_value = 0, acid_value = 0)
+	melee = melee_value
+	bullet = bullet_value
+	laser = laser_value
+	energy = energy_value
+	bomb = bomb_value
+	bio = bio_value
+	rad = rad_value
+	fire = fire_value
+	acid = acid_value
+	tag = ARMORID
+
+/datum/armor/proc/modifyRating(melee_value = 0, bullet_value = 0, laser_value = 0, energy_value = 0, bomb_value = 0, bio_value = 0, rad_value = 0, fire_value = 0, acid_value = 0)
```

Comment:
This is used on the explorer's suit, mining suit, and ripley.

Reply:
All done, I had forgotten to plug them.